### PR TITLE
nrf_rpc: adding OT rpc udp client tests.

### PR DESCRIPTION
**Pull request by canisLupus1313**
Commit adds OT RPC unit tests for UDP client part.

### CODE REVIEW

**Review by bpienk — APPROVED** [review]

File: tests/mocks/nrf_rpc/mock_nrf_rpc_transport.c [hunk 1]
```diff
@@ -85,9 +86,13 @@ static int send(const struct nrf_rpc_tr *transport, const uint8_t *data, size_t
 	expected = &ctx->expected[ctx->cur_expected];
 	response = &ctx->response[ctx->cur_expected];
 	++ctx->cur_expected;
+	log_payload("Expected nRF RPC packet", expected->data, expected->len);
 
-	zassert_equal(expected->len, length, "Unexpected nRF RPC packet sent");
```
Comment (alxelax) [inline tests/mocks/nrf_rpc/mock_nrf_rpc_transport.c:89]: What is the difference. Seems the previous implementation does the same, but in short form.

File: tests/mocks/nrf_rpc/mock_nrf_rpc_transport.c [hunk 2]
```diff
@@ -85,9 +86,13 @@ static int send(const struct nrf_rpc_tr *transport, const uint8_t *data, size_t
...
-	zassert_mem_equal(expected->data, data, length, "Unexpected nRF RPC packet sent");
+	if (expected->len != length) {
+		zexpect(false, "Unexpected nRF RPC packet sent");
```
Comment (alxelax) [inline tests/mocks/nrf_rpc/mock_nrf_rpc_transport.c:92]: Doesn't seem `zexpect` public API. https://docs.zephyrproject.org/apidoc/latest/group__ztest__expect.html

**Review by alxelax — COMMENTED** [review]
LGTM just 2 comments. Please take a look.

File: tests/mocks/nrf_rpc/mock_nrf_rpc_transport.c (hunk 1 above)
Comment (canisLupus1313) [reply tests/mocks/nrf_rpc/mock_nrf_rpc_transport.c:89]: Zassert interrupts the flow of the test by that k_free is not being called. And the expectations in most of the tests also carry between tests.

File: tests/subsys/net/openthread/rpc/client/src/udp_suite.c [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+#include <ot_rpc_types.h>
+#include <test_rpc_env.h>
+
+#include <zephyr/fff.h>
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+
+#include <openthread/udp.h>
+
+/* Fake functions */
```
Comment (alxelax) [inline tests/subsys/net/openthread/rpc/client/src/udp_suite.c:21]: Hm... where?

Comment (canisLupus1313) [reply tests/subsys/net/openthread/rpc/client/src/udp_suite.c:21]: Line 199 :)

**Review by alxelax — APPROVED** [review]

**Pull request merged**